xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailView.swift

```diff
@@ -145,53 +145,31 @@ struct MemoViewerDetailLoadedView: View {
     
     private func onLink(
         url: URL
-    ) -> Bool {
-        guard let sub = url.toSubSlashlinkURL() else {
-            return true
+    ) -> OpenURLAction.Result {
+        guard let link = url.toSubSlashlinkURL() else {
+            return .systemAction
         }
-        
-        // Stitch the base address on to the tapped link
-        // this is needed in the viewer but not the editor
-        // because the editor is (currently) always pointed to
-        // our data.
-        let slashlink = Func.run {
-            guard let basePetname = address.petname else {
-                return sub.slashlink
-            }
-            
-            return sub.slashlink.relativeTo(petname: basePetname)
-        }
-        
```

Comment:
Why was this removed? 

Without this petname links do not work correctly. If I am viewing `@gordon/hello` and tap a link to `@ben` I want `@ben.gordon` but without this code I will just get `@ben`. Only the viewer has this case to worry about, the editor is always starting from your address book as the root but the viewer is relative to the current memo's owner.

Reply:
Oops! Mistaken deletion. Fixed.